presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -196,11 +201,15 @@ private RelationPlan addRowFilters(Table node, RelationPlan plan)
 
     private RelationPlan addColumnMasks(Table table, RelationPlan plan)
     {
+        Map<String, List<Expression>> columnMasks = analysis.getColumnMasks(table);
+
+        if (columnMasks.isEmpty()) {
+            return plan;
+        }
```

Comment:
Why exactly?
This is just short-circuiting for speed?
I am concerned this may reduce robustness of the code below (eg should `initializePlanBuilder(plan)` have possibility of failing, it would not fail, because masks are rare).

if this stays, this should be likewise applied in the `addRowFilters`

Reply:
For WITH clauses, there will be a call to `addColumnMasks` with a map that will be empty (column masks don't apply to WITH clauses, only tables and views). The code below tries to walk over the list of fields from that relation, which is expected to have names as it's supposed to originate from a view or table. In the case of a WITH clause, names may be missing, causing a failure.

For row filters it doesn't matter, as they don't refer to columns by name.

I debated going this route or making the loop below skip fields that don't have a name. There's no case where there can be column masks and some fields not have a name, so it felt unnecessary. I would also likely be a bug if that happened, and the current approach will fail with the expected error if it happens.

I can add a comment to clarify, as it's evidently not obvious why it's there.

Reply:
That deserves a code comment.

However, can we explicitly exclude CTEs from this code?

Reply:
That's hard. At this point of the process there's no difference between a view and a CTE -- they are all just "named queries".

Reply:
Then a comment.